components/suggestion/search_channel_provider.jsx

```diff
@@ -51,10 +56,9 @@ export default class SearchChannelProvider extends Provider {
                         channels = channels.filter((ch) => isDirectChannel(ch) || isGroupChannel(ch));
                     }
 
-                    //
-                    // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
-                    //
-                    channels = channels.sort(sortChannelsByTypeListAndDisplayName.bind(null, 'en', [Constants.OPEN_CHANNEL, Constants.PRIVATE_CHANNEL, Constants.DM_CHANNEL, Constants.GM_CHANNEL]));
+                    const locale = getCurrentLocale(getState());
+
+                    channels = channels.sort(sortChannelsByTypeListAndDisplayName.bind(null, locale, [Constants.OPEN_CHANNEL, Constants.PRIVATE_CHANNEL, Constants.DM_CHANNEL, Constants.GM_CHANNEL]));
```

Comment:
This looks fine but we are adding access to the store here, could we have avoided by passing them into the `handlePretextChanged`?